raiden/raiden_service.py

```diff
@@ -478,21 +483,29 @@ def _initialize_wal(self) -> None:
         storage.update_version()
         storage.log_run()
 
+        initial_state = self.make_initial_state()
+
         try:
             (
+                state_snapshot,
+                state_change_start,
                 state_change_qty_snapshot,
-                state_change_qty_pending,
-                restore_wal,
-            ) = wal.restore_to_state_change(
+            ) = wal.restore_or_init_snapshot(
```

Comment:
This will save the result of `make_initial_state` which is necessary for the replay to be deterministic.